play-routes-compiler-cli/CommandLinePlayRoutesCompiler.scala

```diff
@@ -0,0 +1,166 @@
+package rulesplayroutes.routes
+
+import com.google.devtools.build.buildjar.jarhelper.JarCreator
+import higherkindness.rules_scala.common.error.AnnexWorkerError
+import higherkindness.rules_scala.common.worker.WorkerMain
+import higherkindness.rules_scala.common.sandbox.SandboxUtil
+import java.io.{File, PrintStream}
+import java.nio.file.{Files, Path, Paths}
+import play.routes.compiler._
+import play.routes.compiler.RoutesCompiler.RoutesCompilerTask
+import scala.collection.mutable.ListBuffer
+import scala.jdk.CollectionConverters._
+import scala.io.Source
+import scala.util.{Try, Success, Failure}
+import scopt.Read
+import scala.Console._
+import scopt.OptionParser
+
+object CommandLinePlayRoutesCompiler extends WorkerMain[Unit] {
+  implicit val listPathRead: Read[List[Path]] = Read.reads { commaList =>
+    commaList.split(',').view.map(Paths.get(_)).toList
+  }
+
+  case class Config(
+    var sources: List[Path] = List.empty[Path],
+    var outputDirectory: Path = Paths.get("."),
+    var outputSrcJar: Path = Paths.get("."),
+    var additionalImports: List[String] = List.empty[String],
+    var routesGenerator: RoutesGenerator = InjectedRoutesGenerator,
+    var generateReverseRouter: Boolean = false,
+    var namespaceReverserRouter: Boolean = false,
+    var generateForwardsRouter: Boolean = true
+  )
+
+  def parser(workDir: Path) = new OptionParser[Config]("scopt") {
+    head("Command Line Play Routes Compiler", "0.1")
+
+    arg[Path]("<outputDirectory>").required().action { (outputDirectory, config) =>
+      config.outputDirectory = SandboxUtil.getSandboxPath(workDir, outputDirectory)
```

Comment:
I think copying an object n times to set n options in a ~200 line file hurts performance without nearly any benefit. If this was a larger or more complex codebase, then sure. As it stands, I think leaving this immutable would be pursuing immutable as a goal unto itself and thus lean towards dogmatism rather than pragmatism.

Reply:
Oh, I didn't realize there would be so many options :man_facepalming: 

That makes sense.